src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -36,7 +36,13 @@ public DeleteCommand parse(String args) throws ParseException {
             String trimmedInput = args.trim();
             NameContainsKeywordsPredicate predicate = new NameContainsKeywordsPredicate(List.of(trimmedInput));
             model.updateFilteredPersonList(predicate);
-            return new DeleteCommand(predicate);
+            if (model.getFilteredPersonList().size() == 0) {
+                throw new ParseException("The person / name does not exist ! Try again");
+            } else if (model.getFilteredPersonList().size() > 1) {
+                throw new ParseException("There are more than 1 person in the given list!");
+            } else {
+                return new DeleteCommand(Index.fromOneBased(1));
+            }
```

Comment:
Hi,

For this part, can we use the FindCommand's ability to parse. If we try writing
```
delete
```
with no attached predicates, this will throw illegal argument exceptions. `FindCommand` already accounts for this and tying them together can reduce errors when searching by name.
```suggestion
            new FindCommandParser().parse(args).execute(model);

            if (model.getFilteredPersonList().size() == 0) {
                throw new ParseException(String.format(MESSAGE_INVALID_NAME, args), pe);
            } else if (model.getFilteredPersonList().size() > 1) {
                throw new ParseException(String.format(MESSAGE_INVALID_AMBIGUOUS_NAME, args), pe);
            } else {
                return new DeleteCommand(Index.fromOneBased(1));
            }
```

(with static Messages import at the top)

Reply:
Thanks for the suggestion. Updated with the new implementation. Please check.